From bfd3125066cfc722e244c9f651a3e5dafb8d005d Mon Sep 17 00:00:00 2001
From: Jeremy Allison <jra@samba.org>
Date: Mon, 24 Feb 2014 16:18:31 -0800
Subject: [PATCH] s3-printing: Fix obvious memory leak in
 printer_list_get_printer().

https://bugzilla.samba.org/show_bug.cgi?id=9993

Signed-off-by: Jeremy Allison <jra@samba.org>
Reviewed-by: Ira Cooper <ira@samba.org>
Reviewed-by: Andreas Schneider <asn@samba.org>

Autobuild-User(master): Andreas Schneider <asn@cryptomilk.org>
Autobuild-Date(master): Tue Feb 25 13:19:37 CET 2014 on sn-devel-104

(cherry picked from commit 148bbdd8d04400b5d873f636671dd443952ca04f)
---
 source3/printing/printer_list.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/source3/printing/printer_list.c b/source3/printing/printer_list.c
index 6e02ee5..7e89ec4 100644
--- a/source3/printing/printer_list.c
+++ b/source3/printing/printer_list.c
@@ -133,6 +133,7 @@ NTSTATUS printer_list_get_printer(TALLOC_CTX *mem_ctx,
 done:
 	SAFE_FREE(nstr);
 	SAFE_FREE(cstr);
+	SAFE_FREE(lstr);
 	TALLOC_FREE(key);
 	return status;
 }
-- 
1.8.5.2

